src/dotnet/UI.Blazor.App/Components/ChatList/ChatListItem.razor

```diff
@@ -187,13 +203,52 @@
             IsSelected = chatState.IsSelected,
             UnreadCount = chatState.Info.UnreadCount,
             HasUnreadMentions = chatState.Info.HasUnreadMentions,
-            LastTextEntry = chatState.Info.LastTextEntry,
+            LastTextEntry = lastEntry,
             LastTextEntryText = chatState.Info.LastTextEntryText,
             IsLastItemInBlock = isLastItemInBlock,
             IsFirstItem = isFirstItem,
+            IsUnreadByOthers = isUnreadByOthers,
         };
     }
 
+    private async Task<bool> IsUnreadByOthers(ChatEntry entry, AuthorId ownAuthorId, CancellationToken cancellationToken) {
+        if (_hasBeenReadEntryId == entry.Id)
+            return false;
+
+        var isOwnMessage = !ownAuthorId.IsNone && ownAuthorId == entry.AuthorId;
+        if (!isOwnMessage)
+            return false;
+
+        if (!await ChatSupportsUnreadStatus(cancellationToken).ConfigureAwait(false))
+            return false;
+
+        var readPositionsStat = await Chats.GetReadPositionsStat(Session, entry.ChatId, cancellationToken).ConfigureAwait(false);
+        var canCalculateHasRead = readPositionsStat.CanCalculateHasReadByAnotherAuthor(entry);
+        // Do not show `Unread` label for old messages (before the feature was introduced)
+        // until we have at least more or less good enough read positions stat.
+        if (!canCalculateHasRead)
+            return false;
+
+        var hasBeenRead = readPositionsStat.HasReadByAnotherAuthor(entry, ownAuthorId);
+        if (hasBeenRead)
+            _hasBeenReadEntryId = entry.Id; // Remember that message has been read, so we don't need to recalculate it on next ComputeState cycle.
+        return !hasBeenRead;
+    }
+
+    private async Task<bool> ChatSupportsUnreadStatus(CancellationToken cancellationToken) {
+        // NOTE(DF): move this check to ChatContext to share among all message views.
+        // ChatContext and Chat can be considered as immutable and used from any thread.
+        var chat = Chat;
+        if (chat.HasSingleAuthor)
+            return false;
+
+        var authors = await Authors.ListAuthorIds(Session, chat.Id, cancellationToken).ConfigureAwait(false);
```

Comment:
I think it's not good to fetch author ids for all list items. It may hurt performance. Better to look for another solution.